### PR TITLE
RC-34459: Support cluster override sharing

### PR DESCRIPTION
**Pull request by nikhil-rafay**

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -230,11 +218,12 @@ require (
 
 replace (
 	cloud.google.com/go => cloud.google.com/go v0.100.2
-	// github.com/RafaySystems/rafay-common => ../rafay-common
+	github.com/RafaySystems/rafay-common => ../rafay-common
```
Comment (krishnarafay) [inline go.mod:221]: correct it

File: go.mod [hunk 2]
```diff
@@ -230,11 +218,12 @@ require (
...
 	//github.com/RafaySystems/rctl => github.com/RafaySystems/rctl v1.5.14
 	// github.com/RafaySystems/terraform-provider-rafay/rafay => ../rafay
-	// github.com/RafaySystems/rctl => ../rctl
+	github.com/RafaySystems/rctl => ../rctl
```
Comment (krishnarafay) [inline go.mod:224]: correct it here also.

File: rafay/provider.go [hunk 3]
```diff
@@ -76,6 +76,7 @@ func New(_ string) func() *schema.Provider {
 				"rafay_namespace_mesh_rule":               resourceNamespaceMeshRule(),
 				"rafay_namespace_mesh_policy":             resourceNamespaceMeshPolicy(),
 				"rafay_cluster_sharing":                   resourceClusterSharing(),
+				"rafay_cluster_override_sharing":          resourceClusterOverrideSharing(),
```
Comment (krishnarafay) [inline rafay/provider.go:79]: I think we are not using this now..

File: rafay/rafay_cluster_override_sharing.go [hunk 4]
```diff
@@ -0,0 +1,516 @@
+package rafay
```
Comment (krishnarafay) [inline rafay/rafay_cluster_override_sharing.go:1]: This file `rafay_cluster_override_sharing.go` may be not needed. we can delete right..?

**Comment by krishnarafay** [comment]
can you update the document ( md files ), similar to this PR https://github.com/RafaySystems/terraform-provider-rafay/pull/586/files#diff-05368d31aba33b5696383579d564a172f45b7d8f9f0fbd9ae160c69fb75f3fe0 .

**Review by krishnarafay — APPROVED** [review]

**Review by rafay-vivek — APPROVED** [review]

**Review by krishnarafay — APPROVED** [review]

**Review by janardhan-rafay — APPROVED** [review]

**Pull request merged**